aquarius/events/events_monitor.py

```diff
@@ -574,70 +505,105 @@ def get_assets_in_chain(self):
-        elif event_name == EventTypes.EVENT_METADATA_STATE:
-            hash_text = "MetadataState(address,uint8,uint256,uint256)"
-        elif event_name == EventTypes.EVENT_TOKEN_URI_UPDATE:
-            hash_text = "TokenURIUpdate(address,string,uint256,uint256,uint256)"
-        elif event_name == EventTypes.EVENT_EXCHANGE_CREATED:
-            hash_text = "ExchangeCreated(bytes32,address,address,address,uint256)"
-        elif event_name == EventTypes.EVENT_EXCHANGE_RATE_CHANGED:
-            hash_text = "ExchangeRateChanged(bytes32,address,uint256)"
-        elif event_name == EventTypes.EVENT_DISPENSER_CREATED:
-            hash_text = "DispenserCreated(address,address,uint256,uint256,address)"
-        elif event_name == EventTypes.EVENT_TRANSFER:
-            hash_text = "Transfer(address,address,uint256)"
-        else:
-            hash_text = (
-                "OrderStarted(address,address,uint256,uint256,uint256,address,uint256)"
-            )
-
-        event_signature_hash = self._web3.keccak(text=hash_text).hex()
-
-        _from = from_block
-        _to = min(_from + chunk_size - 1, to_block)
-
-        logger.debug(
-            f"Searching for {event_name} events on chain {self._chain_id} "
+    def get_event_logs(self, from_block, to_block):
+        """Get all events from -> to and process them.
+        If that fails, and we tried with multiple blocks, let split handle it
+        If that fails, and we tried on a single block, then try to get events one by one instead of all
+        """
+        logger.info(
+            f"Searching for events events on chain {self._chain_id} "
             f"in blocks {from_block} to {to_block}."
         )
 
         filter_params = {
-            "topics": [event_signature_hash],
-            "fromBlock": _from,
-            "toBlock": _to,
+            "topics": [list(EventTypes.hashes.keys())],
+            "fromBlock": from_block,
+            "toBlock": to_block,
         }
 
-        all_logs = []
-        while _from <= to_block:
-            # Search current chunk
+        try:
             logs = self._web3.eth.get_logs(filter_params)
-            all_logs.extend(logs)
-            if (_from - from_block) % 1000 == 0:
-                logger.debug(
-                    f"Searched blocks {_from} to {_to} on chain {self._chain_id}; "
-                    f"{len(all_logs)} {event_name} events detected so far."
-                )
+        except Exception as e:
+            if from_block < to_block:
+                # splitting in two might help, so rely on that
+                raise Exception(f"Failed to get events for multiple blocks")
+            else:
+                # Since there is only one block, and we failed to get all events, we need to try to take them one by one
+                # if any call fails, there is nothing more we can do  (ie:  failed to get only transfer events from block X)
+                for topic in EventTypes.hashes:
```

Comment:
The logic is entire different:
1. in order to reduce no of web3 calls, we do eth.get_logs with multiple topics  -> pure performance 99.9% of the time, when in a chunk of blocks, we do get all events, of all types in one single call
2. but if that call fails, we have 2 options:
2a  - if it was a chunk, divide it in two -> then with only 2 web3 calls, we get all events (see 1)
2b - if it was a single block, then it means we cannot fetch all events.  Then we need to do it event by event (just like old aqua was doing).  This should be the last resort, because it wastes a lot of calls

Reply:
yes, I just mean to cleanup the duplications in the code. Keep the calls the same as they are, just route them through a single function for better clarity.

Reply:
done in https://github.com/oceanprotocol/aquarius/pull/928/commits/da5ef9d1fd9414655407c535231c5906a62e83f4